src/Services/NodeUptimeNotificationService.ts

```diff
@@ -0,0 +1,79 @@
+import {NodeUptime} from "../Models/NodeUptime";
+import {EmailService} from "./EmailService";
+import logger from "./Logger";
+import {UserService} from "./UserService";
+import {NodeService} from "./NodeService";
+import {NodeStatusService} from "./NodeStatusService";
+import config from "../Config/Config";
+
+export class NodeUptimeNotificationService {
+
+    private emailService: EmailService;
+    private userService: UserService;
+    private nodeService: NodeService;
+    private nodeStatusService: NodeStatusService;
+
+    constructor(
+        emailService: EmailService,
+        userService: UserService,
+        nodeService: NodeService,
+        nodeStatusService: NodeStatusService
+    ) {
+        this.emailService = emailService;
+        this.userService = userService;
+        this.nodeService = nodeService;
+        this.nodeStatusService = nodeStatusService
+    }
+
+    public async processNodeUptime(nodeUptime: NodeUptime): Promise<void> {
```

Comment:
Maybe split this logic into more functions that have meaningful name? Took me a while to realize what it does